pkg/validation/cluster.go

```diff
@@ -302,26 +302,27 @@ func validateOpenStackCloudSpec(spec *kubermaticv1.OpenstackCloudSpec, dc *kuber
 			return err
 		}
 	}
-	if spec.Username == "" {
-		if err := kuberneteshelper.ValidateSecretKeySelector(spec.CredentialsReference, resources.OpenstackUsername); err != nil {
-			return err
+	if spec.ApplicationCredentialID == "" || spec.ApplicationCredentialSecret == "" {
```

Comment:
This is a simplified check for the Application ID/Secret.  I think this should be fine too but can you check?